### PR TITLE
TEL-5209 TEL-5195 TEL-5190 Changes to AMR-WB

### PR DESCRIPTION
**Pull request by piotrgregor**
- Add app to print the module config (always print it on module's load)
- Improve codec negotiation according to invite-prefer-oa/invite-prefer-be
- Support new XML config params:
 
```
  default-bitrate
 		Bitrate mode that will be used if mode-set-overwrite 
                and mode-set-overwrite-with-default-bitrate are set).
  volte
 		If set, configures codec for use on cellular networks.
  adjust-bitrate
 		Vary bitrate according to feedback from RTCP.
  force-oa
 		Configure codec in octet aligned mode.
  force-be
 		Configure codec in bandwidth efficient mode.
  mode-set-overwrite
 		When answering a call, use codec bitrate modes from mode-set param, 
                instead of mirroring the OFFER.
  mode-set-overwrite-with-default-bitrate
 		If mode-set-overwrite is on, then use default-bitrate mode 
                instead of mode-set.
  invite-prefer-oa
 		When answering a call, if AMR-WB is offered in 2 modes (octet aligned 
                and bandwidth efficient), select octet aligned.
  invite-prefer-be
 		When answering a call, if AMR-WB is offered in 2 modes (octet aligned 
                and bandwidth efficient), select bandwidth efficient.
  mode-set
 		Provides bitrate modes to be used with mode-set-overwrite 
                (if mode-set-overwrite-with-default-bitrate is off).
  debug
 		If on, print extra codec info (CMR, ToC, last frame flag) 
                at the FS's DEBUG level.
  fmtp-extra
 		Append any extra info to fmtp entry for AMR-WB.
```
- Push new XML config to conf/vanilla

### CODE REVIEW

File: src/mod/codecs/mod_amrwb/mod_amrwb.c [hunk 1]
```diff
@@ -587,18 +772,37 @@ SWITCH_MODULE_LOAD_FUNCTION(mod_amrwb_load)
...
+					globals.force_be = (switch_byte_t) atoi(val);
+				}
 				if (!strcasecmp(var, "mode-set-overwrite")) {
 					globals.mode_set_overwrite = (switch_byte_t) atoi(val);
 				}
+				if (!strcasecmp(var, "mode-set-overwrite-with-default-bitrate")) {
+					globals.mode_set_overwrite_with_default_bitrate = (switch_byte_t) atoi(val);
+				}
+				if (!strcasecmp(var, "invite-prefer-oa")) {
+					globals.invite_prefer_oa = (switch_byte_t) atoi(val);
```
Comment (dev-ryanc) [inline src/mod/codecs/mod_amrwb/mod_amrwb.c:785]: Should we have a default preference if invite_prefer_oa or invite_prefer_be is not set? Otherwise this [line](https://github.com/team-telnyx/freeswitch/pull/167/files#diff-70ccf3d2216fcb3f902808582faa6d967dd7dfe8f0f41f3cacd985cb9982b9e0R261) will always return false

**Review by dev-ryanc — CHANGES_REQUESTED** [review]

File: src/mod/codecs/mod_amrwb/mod_amrwb.c (hunk 1 above)
Comment (piotrgregor) [reply src/mod/codecs/mod_amrwb/mod_amrwb.c:785]: Thanks @dev-ryanc These are set to 0 by default, that means feature is not used (you got no preference). 

I added these params as a way of expressing that you have a preference, and if you don't (both `invite_prefer_oa` or `invite_prefer_be` are not set) then this means you do not prefer either of them - in this case codec selection code will work in default mode, the whole prefer thing is not used. This means (a default) that first AMR-WB from SDP offer will be selected.
In other words we can say that default is select first that is listed on SDP (when both `invite_prefer_oa` or `invite_prefer_be` are not set).

Setting `invite_prefer_oa` or `invite_prefer_be` to 0 is same as not setting them at all (these are flippers to turn feature on/off).

Comment (dev-ryanc) [reply src/mod/codecs/mod_amrwb/mod_amrwb.c:785]: Got it

**Review by dev-ryanc — APPROVED** [review]
